Assets/Scripts/StandardSamples/Services/PlayerDataStorageService.cs

```diff
@@ -0,0 +1,468 @@
+                        // and compose a list of the file names
+                        for (uint fileIndex = 0; fileIndex < callbackInfo.FileCount; fileIndex++)
+                        {
+                            CopyFileMetadataAtIndexOptions options = new CopyFileMetadataAtIndexOptions()
+                            {
+                                LocalUserId = userId,
+                                Index = fileIndex
+                            };
+
+                            Result result = playerStorageHandle.CopyFileMetadataAtIndex(ref options, out FileMetadata? fileMetadata);
+
+                            if (result != Result.Success)
+                            {
+                                tcs.SetException(new Exception($"{nameof(PlayerDataStorageService)} ({nameof(DownloadFile)}): CopyFileMetadataAtIndex returned error. Result code {result}"));
+                                return;
+                            }
+                            else if (fileMetadata != null)
+                            {
+                                if (!string.IsNullOrEmpty(fileMetadata?.Filename))
+                                {
+                                    fileNames.Add(fileMetadata?.Filename);
+                                }
+                            }
+                        }
+
+                        tcs.SetResult(fileNames);
+                    }
+                    finally
+                    {
+                        _playerDataStorageSemaphore.Release();
+                    }
+                });
+            });
+
+            return tcs.Task;
+        }
+
+        /// <summary>
+        /// Starts a download from the player data storage.
+        /// This version calls <see cref="DownloadFile(ProductUserId, string)"/>,
+        /// and simplifies the return down to just an awaitable Task if the wrapper class data is not needed.
+        /// If an exception occurs, the returned task will propagate that exception.
+        /// </summary>
+        /// <param name="userId">The user id that owns the player data storage file to download.</param>
+        /// <param name="fileName">The name of the file to download.</param>
+        /// <returns>An awaitable task containing either the result or an exception.</returns>
+        public async Task<byte[]> DownloadFileAsync(ProductUserId userId, string fileName)
+        {
+            EOSPlayerDataStorageTransferTask transferTask = DownloadFile(userId, fileName);
+
+            await transferTask.InnerTaskCompletionSource.Task;
+
+            if (transferTask.InnerTaskCompletionSource.Task.IsFaulted)
+            {
+                return await Task<byte[]>.FromException<byte[]>(new AggregateException($"{nameof(PlayerDataStorageService)} ({nameof(DownloadFileAsync)}): Failed to download file. Result code {transferTask.ResultCode}", transferTask.InnerTaskCompletionSource.Task.Exception));
```

Comment:
Why are you generating a new task here? If `IsFaulted` is true, that should be just as "catchable" as creating a new one. It's not clear to me what is being gained by this approach.